File: src/main/java/org/kite9/diagram/visualization/planarization/ordering/VertexEdgeOrdering.java
package org.kite9.diagram.visualization.planarization.ordering;

import org.kite9.diagram.common.elements.edge.PlanarizationEdge;
import org.kite9.diagram.model.position.Direction;

/**
 * Defines the ordering of edges around a vertex and provides methods for changing the order.
 * 
 * @author robmoffat
 *
 */
public interface VertexEdgeOrdering extends EdgeOrdering {

	public void addEdgeDirection(Direction d, boolean isContradicting);

	public void remove(PlanarizationEdge toRemove);

	public void replace(PlanarizationEdge b, PlanarizationEdge a);
	
}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/ordering/VertexEdgeOrdering.kt
package org.kite9.diagram.visualization.planarization.ordering

import org.kite9.diagram.common.elements.edge.PlanarizationEdge
import org.kite9.diagram.model.position.Direction

/**
 * Defines the ordering of edges around a vertex and provides methods for changing the order.
 *
 * @author robmoffat
 */
interface VertexEdgeOrdering : EdgeOrdering {

    fun addEdgeDirection(d: Direction, isContradicting: Boolean)
    fun remove(toRemove: PlanarizationEdge)
    fun replace(b: PlanarizationEdge, a: PlanarizationEdge)
}